File: client/src/app/shared/models/archive-image.ts
export interface ArchiveImage {
  id: number;
  imageUrl: string;
  title: string;
  description?: string;
  displayOrder: number;
  isMainImage: boolean;
  objectPositionX?: number; // Optional because API might not always return it
  objectPositionY?: number; // Optional because API might not always return it
  createdAt: string;
}

export interface CreateArchiveImageDto {
  imageUrl: string;
  title: string;
  description?: string;
  displayOrder: number;
  isMainImage: boolean;
  objectPositionX: number;
  objectPositionY: number;
}

export interface UpdateArchiveImageDto {
  imageUrl?: string;
  title?: string;
  description?: string;
  displayOrder?: number;
  isMainImage?: boolean;
  objectPositionX?: number;
  objectPositionY?: number;
}

File: Core/Entities/ArchiveImage.cs
namespace Core.Entities;

public class ArchiveImage : BaseEntity
{
    public required string ImageUrl { get; set; }
    public required string Title { get; set; }
    public string? Description { get; set; }
    public int DisplayOrder { get; set; }
    public bool IsMainImage { get; set; }
    public double ObjectPositionX { get; set; } = 50; // Percentage for CSS object-position (default: 50% center)
    public double ObjectPositionY { get; set; } = 50; // Percentage for CSS object-position (default: 50% center)
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

